quadriga/src/main/java/edu/asu/diging/quadriga/core/service/impl/MappedCollectionServiceImpl.java

```diff
@@ -0,0 +1,164 @@
+package edu.asu.diging.quadriga.core.service.impl;
+
+import org.bson.types.ObjectId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.quadriga.core.data.MappedCollectionRepository;
+import edu.asu.diging.quadriga.core.exceptions.CollectionNotFoundException;
+import edu.asu.diging.quadriga.core.exceptions.InvalidObjectIdException;
+import edu.asu.diging.quadriga.core.exceptions.MappedCollectionNotFoundException;
+import edu.asu.diging.quadriga.core.model.Collection;
+import edu.asu.diging.quadriga.core.model.MappedCollection;
+import edu.asu.diging.quadriga.core.service.CollectionManager;
+import edu.asu.diging.quadriga.core.service.MappedCollectionService;
+
+@Service
+public class MappedCollectionServiceImpl implements MappedCollectionService {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private MappedCollectionRepository mappedCollectionRepository;
+
+    @Autowired
+    private CollectionManager collectionManager;
+
+    /**
+     * This method checks whether a collection with given collectionId exists and
+     * returns the collection if it exists If one doesn't exist, then the
+     * collectionId shouldn't be linked to the mappedCollection
+     * 
+     * @param collectionId is the id of the collection to be checked
+     * @return the Collection entry found in the database
+     * @throws InvalidObjectIdException    if collectionId couldn't be conveted to
+     *                                     ObjectId
+     * @throws CollectionNotFoundException if collection with given collectionId
+     *                                     does't exist
+     */
+    private Collection checkAndGetCollection(String collectionId)
```

Comment:
There was a comment here before, right? why don't you use the collection manager and then check for null instead of creating its own method? This seems unnecessarily complicated.

Reply:
This is actually an old PR that has not been updated since September 30. Instead of re-opening a PR for this story, I had created a new one and you asked me to close it, so this is the closed one

On the PR that is currently under review by you (https://github.com/diging/quadriga-2.0/pull/17), I have removed this method